doc/provisioning.md

```diff
@@ -1,19 +1,28 @@
-What We Run, and Why
-====================
-
-Expressed as compactly as possible, the Provisioner is responsible for making
-sure that non-Dead machine entities in state have agents running on live
-instances; and for making sure that Dead machines, and stray instances, are
-removed and cleaned up.
-
-However, the choice of exactly what we deploy involves some subtleties. At the
-Provisioner level, it's simple: the base and the constraints we pass to the
-Environ.StartInstance come from the machine entity. But how did they get there?
-
-Bases
------
-
-Individual charms are released for different possible target base; juju
+# Provisioning
+
+Provisioning in Juju is performed by two separate workers:
+- The [compute provisioner](https://github.com/juju/juju/blob/main/internal/worker/computeprovisioner/doc.go),
+which is responsible for provisioning machine instances on the underlying
+provider.
+- The [container provisioner](https://github.com/juju/juju/blob/main/internal/worker/containerprovisioner/doc.go),
+which is responsible for creating containers managed by the container
+broker on the underlying machine.
+
+Juju splits the resource creation in two stages. When a new machine
+is being created (either by the user explictly or because a new unit
+is deployed) the machine is first stored in the database, and then
+one of the provisioners (compute or container) workers will pick it
+up and start the provisioning process. Both workers will gather all the
+necessary information and then create a [provisioner_task](https://github.com/juju/juju/blob/main/internal/provisionertask/provisioner_task.go)
+worker which contains all the bussiness logic for creating and updating
```

Comment:
Maybe split this into a second sentence from the "which".

Reply:
Yeah you're right, fixed.